### ISSUE TITLE
[TD-91] taskd crashes with core dump if CRL file is malformed

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskserver#91]
_Jens Erat on 2015-02-27T14:48:54Z says:_ 

 taskd 1.0 crashes with core dump if the certificate revocation list is malformed.

Current behavior:

{code}
$ taskd config server.crl /dev/null # or any non-PEM file
$ taskd server

*** Error in `/usr/local/bin/taskd': free(): invalid pointer: 0x000000000042224a ***
{code}

Expected behavior: either work without crl, or print a reasonable error message

Diagnostic information:

{code}
1.  taskd diagnostics

taskd 1.0.0
    Platform: Linux
    Hostname: c6e6be2ee681

Compiler
     Version: 4.9.1
        Caps: +stdc +stdc_hosted +200809 +LP64 +c1 +i4 +l8 +vp8

Build Features
       Built: Feb 27 2015 13:03:22
       CMake: 3.0.2
        Caps: +pthreads +tls
     libuuid: libuuid + uuid_unparse_lower
   libgnutls: 3.3.8
{code}

Probably barely helpful stack trace without debug symbols:

{code}
(gdb) bt
1. 0  0x00007f1cf9c5d107 in __GI_raise (sig=sig(at)entry=6)
    at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
1. 1  0x00007f1cf9c5e4e8 in __GI_abort () at abort.c:89
1. 2  0x00007f1cf9c9b044 in __libc_message (do_abort=do_abort(at)entry=1, 
    fmt=fmt(at)entry=0x7f1cf9d8dc60 "*** Error in `%s': %s: 0x%s ***\n")
    at ../sysdeps/posix/libc_fatal.c:175
1. 3  0x00007f1cf9ca081e in malloc_printerr (action=1, 
    str=0x7f1cf9d89d1e "free(): invalid pointer", ptr=<optimized out>) at malloc.c:4996
1. 4  0x00007f1cf9ca1526 in _int_free (av=<optimized out>, p=<optimized out>, have_lock=0)
    at malloc.c:3840
1. 5  0x000000000046639f in TLSServer::~TLSServer() ()
1. 6  0x000000000045734a in Server::beginServer() ()
1. 7  0x000000000043a4b8 in command_server(Database&, std::vector<std::string, std::allocator<std::string> > const&) ()
1. 8  0x0000000000420352 in main ()
{code}

I did not verify against git HEAD yet, but I couldn't find any related bug reports.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2015-02-27T14:48:54Z
Modified: 2015-02-27T23:18:16Z
```

**Comment by taskwarrior** [comment]
_Jens Erat on 2015-02-27T15:24:21Z says:_ 

 Possibly indeed resolved:
*  CRL cert is now optional (thanks to Anton Khirnov).

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2015-02-27T23:18:16Z says:_ 

 You are right, this is fixed in 1.1.0, to be released soon.  Soonish.
